at_client/lib/src/manager/at_client_manager.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:at_client/at_client.dart';
+import 'package:at_client/src/listener/at_sign_change_listener.dart';
+
+class AtClientManager {
+  var _atSign;
+  var currenAtClient;
+  final _changeListeners = <AtSignChangeListener>[];
+
+  static final AtClientManager _singleton = AtClientManager._internal();
+
+  AtClientManager._internal();
+
+  factory AtClientManager.getInstance() {
+    return _singleton;
+  }
+
+  void setCurrentAtSign(
+      String atSign, String namespace, AtClientPreference preference) {
+    _atSign = _atSign;
+    currenAtClient = AtClientImpl(_atSign, namespace, preference);
```

Comment:
I believe the map inside should ensure that for the same @sign we don't create multiple instances of AtClient